transport/src/main/java/io/netty/channel/SubChannel.java

```diff
@@ -0,0 +1,520 @@
+package io.netty.channel;
+
+import java.net.SocketAddress;
+import java.util.concurrent.RejectedExecutionException;
+
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.RecvByteBufAllocator.Handle;
+import io.netty.util.Attribute;
+import io.netty.util.AttributeKey;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+public class SubChannel implements Channel {
+
+	private static final InternalLogger logger = InternalLoggerFactory.getInstance(SubChannel.class);
+
+	private final class SubChannelId implements ChannelId {
+
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public int compareTo(ChannelId o) {
+			return parent().id().compareTo(o);
+		}
+
+		@Override
+		public String asShortText() {
+			return parent().id().asShortText();
+		}
+
+		@Override
+		public String asLongText() {
+			return parent().id().asLongText();
+		}
+
+	}
+
+	private final SubChannelUnsafe unsafe = new SubChannelUnsafe();
+	private final ChannelId channelId;
+	private final ChannelPipeline pipeline;
+
+	protected final ChannelHandlerContext ctx;
+
+	private volatile boolean registered;
+
+	public SubChannel(final ChannelHandlerContext ctx) {
+		this.ctx = ctx;
+		pipeline = new DefaultChannelPipeline(this) {
```

Comment:
Should this be a standalone class?

Reply:
Perhaps. Do you mean, a standalone private static class within the same file, rather than an anonymous class? If so, probably!